cpp/src/strings/convert/utilities.cuh

```diff
@@ -0,0 +1,151 @@
+  size_type digits =
+    (value < 10
+       ? 1
+       : (value < 100
+            ? 2
+            : (value < 1000
+                 ? 3
+                 : (value < 10000
+                      ? 4
+                      : (value < 100000
+                           ? 5
+                           : (value < 1000000
+                                ? 6
+                                : (value < 10000000
+                                     ? 7
+                                     : (value < 100000000
+                                          ? 8
+                                          : (value < 1000000000
+                                               ? 9
+                                               : (value < 10000000000
+                                                    ? 10
+                                                    : (value < 100000000000
+                                                         ? 11
+                                                         : (value < 1000000000000
+                                                              ? 12
+                                                              : (value < 10000000000000
+                                                                   ? 13
+                                                                   : (value < 100000000000000
+                                                                        ? 14
+                                                                        : (value < 1000000000000000
+                                                                             ? 15
+                                                                             : (value <
+                                                                                    10000000000000000
+                                                                                  ? 16
+                                                                                  : (value <
+                                                                                         100000000000000000
+                                                                                       ? 17
+                                                                                       : (value <
+                                                                                              1000000000000000000
+                                                                                            ? 18
+                                                                                            : (value <
+                                                                                                   10000000000000000000
+                                                                                                 ? 19
+                                                                                                 : 20)))))))))))))))))));
```

Comment:
Please turn clang-format off around this cascade. It makes it much tidier:

```suggestion
  // clang-format off
  size_type digits =
    (value < 10 ? 1 :   
    (value < 100 ? 2 :   
    (value < 1000 ? 3 :   
    (value < 10000 ? 4 :   
    ...
    ...))));
    // clang-format on
```

Make sure to get the whitespace right (none at end of line) `// clang-format off`.

Reply:
done